### PR TITLE
Fix examples/translation/t5 to use newstest2014 rather than newstest2013

### LINKED ISSUE #3759: Why does `examples/translation/t5` test on newstest2013 rather than newstest2014?
# Details

The example in examples/translation/t5 uses `newstest2013`, but authors report against `newstest2014` (presumably newstest2014.full):
> Since this is our final set of experiments, we report results on the test set rather than
the validation set. For CNN/Daily Mail, we use the standard test set distributed with the dataset.
For the WMT tasks, this corresponds to using newstest2014 for English-German
[Original paper, p30](https://arxiv.org/pdf/1910.10683.pdf).

Is this intentional?

### PR DESCRIPTION
**Pull request by tholiao**
Resolves #3759, in addition to minor nits: fixed a bug with argparse arguments + more pythonic file handling + formatted with black and isort. 

Please note that I have not yet run the evaluation script against the full newstest2014 test set, as it is rather compute intensive, so the disclaimer at the top of the README.md about the score gap between the pre-trained and fine-tuned models is only ostensibly accurate to the score gap on newstest2013, not newstest2014.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Hey @tholiao, 

Thanks a lot for the PR :-) This looks good so far. Don't worry about running the script against the evaluation set - we can do this once this is merged! 

Can you make sure that the `run_examples_torch` pass? Don't worry too much about the `check_code_quality` test - there have been some issues with `isort` and I can manually fix that later.

**Comment by tholiao** [comment]
Should be fine now.

File: examples/translation/t5/evaluate_wmt.py [hunk 1]
```diff
@@ -27,67 +25,104 @@ def generate_translations(lns, output_file_path, model_size, batch_size, device)
...
-    score_file.write(result)
+    with Path(score_path).open("w") as score_file:
+        score_file.write(result)
 
 
 def run_generate():
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "model_size",
+        "--model_size",
```
Comment (patrickvonplaten) [inline examples/translation/t5/evaluate_wmt.py:63]: Can you leave that as just `model_size`?

File: examples/translation/t5/evaluate_wmt.py [hunk 2]
```diff
@@ -27,67 +25,104 @@ def generate_translations(lns, output_file_path, model_size, batch_size, device)
...
 
     args = parser.parse_args()
-    args.device = torch.device("cuda" if torch.cuda.is_available() and not args.no_cuda else "cpu")
+    args.device = torch.device(
+        "cuda" if torch.cuda.is_available() and not args.no_cuda else "cpu"
+    )
 
     dash_pattern = (" ##AT##-##AT## ", "-")
 
-    input_lns = [x.strip().replace(dash_pattern[0], dash_pattern[1]) for x in open(args.input_path).readlines()]
```
Comment (patrickvonplaten) [inline examples/translation/t5/evaluate_wmt.py:85]: It's better to delete the three empty lists IMO, otherwise, we just propagate a silent error

File: examples/translation/t5/test_t5_examples.py [hunk 3]
```diff
@@ -37,6 +41,7 @@ def test_t5_cli(self):
 
         testargs = [
             "evaluate_wmt.py",
+            "--model_size",
```
Comment (patrickvonplaten) [inline examples/translation/t5/test_t5_examples.py:44]: delete `--model_size` arg should work without.

**Comment by patrickvonplaten** [comment]
Hi @tholiao, I went into your PR branch and checked the `make style` issues. It seems like you have other params set up for `black` than this lib. Also `isort` seems to have some problems with the imports here. 

I added the tiny changes, I suggested above and correctly formatted everything (black and isort) in this PR https://github.com/huggingface/transformers/pull/3817. Since you did all the work and I don't wanna take the main authorship, I suggest that you either copy / paste the files I changed from my PR branch into yours or you cherry pick the 4 commits are put on top of yours to your PR. 

After this I think we are good to merge :-)
